src/browser/api_protocol/api_handlers/event_listener.js

```diff
@@ -87,6 +85,42 @@ function EventListenerApiHandler() {
                 };
             }
         },
+        'frame': {
+            name: 'frame',
+            subscribe: function(identity, type, payload, cb) {
+                const {
+                    uuid,
+                    name
+                } = payload;
+                const frameIdentity = apiProtocolBase.getTargetWindowIdentity(payload);
+                const targetUuid = frameIdentity.uuid;
+                const islocalWindow = !!coreState.getWindowByUuidName(targetUuid, targetUuid);
+                const localUnsub = Frame.addEventListener(identity, frameIdentity, type, cb);
+                let remoteUnSub;
```

Comment:
Can this be moved down closer to the conditional where it's assigned?

Reply:
yeah, but this is basically a copy/past of the other subscription providers around it. I think they just all be "fixed" en mass (made a factory function or the like) at some other time